src/inline-assembly.md

```diff
@@ -37,515 +52,937 @@ assert_eq!(x, 4 * 6);
+```
+
+r[asm.invocation.positional]
+A `format_specifier` that does not specify an `operand_specifier` is called a positional specifier, and refers to the `nth` successive positional operand, where `n` is `0` for the first positional specifier in the *joined asm-string* and increases by 1 for each successive positional specifier in the *joined asm-string*.
+
+```rust
+# #[cfg(target_arch = "x86_64")] { unsafe{
+let x: i32;
+core::arch::asm!("mov rax, {}", in(reg) 5i64, out("eax") x);
+# }}
+```
+
+r[asm.invocation.explicit-positional]
+A `format_specifier` that has an `operand_specifier` which is a DEC_LITERAL is called an explicit positional specifier, and refers to the `nth` successive positional operand, where `n` is the value of the DEC_LITERAL.
+
+```rust
+# #[cfg(target_arch = "x86_64")] { unsafe{
+let x: i32;
+core::arch::asm!("mov {1}, {0}", in(reg) 5i64, out(reg) x);
+# }}
+```
+
+r[asm.invocation.named]
+A `format_specifier` that has an `operand_specifier` which is an ident is called a named specifier, and refers to the named operand with the specified name.
+
+```rust
+# #[cfg(target_arch = "x86_64")] { unsafe{
+let x: i32;
+core::arch::asm!("mov {output}, {input}", input = in(reg) 5i64, output = out(reg) x);
+# }}
+```
+
+r[asm.invocation.expansion]
+If the `raw` option is not specified, the *joined asm-string* is expanded as defined in [asm.operands.expansion], replacing each `format_specifier` with the appropriate expansion for the operand. The resulting string is called the *expanded asm-string*. If the `raw` option is specified, the *expanded asm-string* is the *joined asm-string* verbatim. 
+
+r[asm.invocation.syntax]
+The syntax of the *expanded asm-string* is a subset of the GNU AS syntax for the target. Invoking the macro with a *expanded asm-string* that does not match syntax requirements is *conditionally supported* and has *assembler dependent behaviour*. Invoking a directive that is not specified by [asm.directives] is *conditionally supported* and has *assembler dependent behaviour*.
+
+> [!TARGET-SPECIFIC]
+> On x86 and x86_64 targets, the syntax of the *expanded asm-string* acts as though the directive `.intel_syntax noprefix` is issued before parsing the *expanded asm-string*, except that the `option(att_syntax)` causes the syntax to act as though the directive `.att_syntax prefix` is issued before parsing the *expanded asm-string* instead.
+> On ARM targets, the syntax of the *expanded asm-string* acts as though the directive `.syntax unified` is issued before parsing the *expanded asm-string*.
+
+r[asm.invocation.duplication]
+The number of times, locations, and the order in which a given invocation of [`core::arch::asm!`] is expanded is unspecified.
+
+```rust,ignore
+// The following code may have suprising results, and may fail to compile or link. 
+// The results, including whether it succesfully compiles, may depend on non-local use sites of the function, and on optimization settings.
```

Comment:
Can you make sure that the code examples are wrapped roughly at 80 columns? The way it gets presented in the book requires horizontal scrolling which can be awkward.

Also, there are some misspellings here.

```suggestion
// The following code may have surprising results, and may fail to compile or
// link. The results, including whether it successfully compiles, may depend on
// non-local use sites of the function, and on optimization settings.
```